cmd/helm/flags.go

```diff
@@ -52,10 +53,20 @@ func addChartPathOptionsFlags(f *pflag.FlagSet, c *action.ChartPathOptions) {
 // bindOutputFlag will add the output flag to the given command and bind the
 // value to the given format pointer
 func bindOutputFlag(cmd *cobra.Command, varRef *output.Format) {
-	cmd.Flags().VarP(newOutputValue(output.Table, varRef), outputFlag, "o",
+	f := cmd.Flags()
+	f.VarP(newOutputValue(output.Table, varRef), outputFlag, "o",
 		fmt.Sprintf("prints the output in the specified format. Allowed values: %s", strings.Join(output.Formats(), ", ")))
-	// Setup shell completion for the flag
-	cmd.MarkFlagCustom(outputFlag, "__helm_output_options")
+
+	flag := f.Lookup(outputFlag)
```

Comment:
instead of creating with `f.VarP` then performing a lookup, have you looked into constructing the flags before-hand?

```
oFlag := pflag.VarP(newOutputValue(output.Table, varRef), outputFlag, "o", fmt.Sprintf("prints the output in the specified format. Allowed values: %s", strings.Join(output.Formats(), ", ")))

f.AddFlag(oFlag)
completion.RegisterFlagCompletionFunc(oFlag, ...
```

Reply:
I assume you mean `pflag.VarPF(...)`.
That would be more efficient. I'll make the change.

Background:
This particular flag is special because it calls `pflag.VarP(...)` directly so could call `pflag.VarPF(...)` instead.

I had hesitantly chosen against it because that technique is not applicable to any of the other helm flags which call more specific methods such as `pflag.IntVar(..)` or `pflag.StringVarP(..)`, and those methods don't have a version that returns the flag.  And since the `--output` flag was the first one I coded in Go, I wanted to have it be an example of how other flags should be coded.

But that now that completion for the other flags are coded, that decision is not really as valid.